ampersand-view-switcher.js

```diff
@@ -31,8 +31,12 @@ ViewSwitcher.prototype.set = function (view) {
     }
 
     if (this.config.waitForRemove) {
+        this.next = view;
         this._hide(prev, function () {
-            self._show(view);
+            if(self.next===view){
+                self.next = null;
```

Comment:
You can delete this property completely here `delete self.next`. Also it'd be great if you could fix code styling (spacing for `if` statement).